rct229/rules/section21/section21rule5.py

```diff
@@ -0,0 +1,161 @@
+            list_path="ruleset_model_instances[*]",
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    class RulesetModelInstanceRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section21Rule5.RulesetModelInstanceRule, self,).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+            )
+
+        def is_applicable(self, context, data=None):
+            rmi_b = context.baseline
+            baseline_system_types_dict = get_baseline_system_types(rmi_b)
+            # create a list contains all HVAC systems that are modeled in the rmi_b
+            available_type_lists = [
+                hvac_type
+                for hvac_type in baseline_system_types_dict.keys()
+                if len(baseline_system_types_dict[hvac_type]) > 0
+            ]
+            return any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_lists
+                ]
+            )
+
+        def get_calc_vals(self, context, data=None):
+            rmi_b = context.baseline
+            climate_zone = data["climate_zone"]
+
+            # get zone conditions from buildings
+            zone_conditioning_category_dict = {}
+            for bldg in find_all("$.buildings[*]", rmi_b):
+                zone_conditioning_category_dict = {
+                    **zone_conditioning_category_dict,
+                    **get_zone_conditioning_category_dict(climate_zone, bldg),
+                }
+
+            loop_zone_list_w_area_dict = get_hw_loop_zone_list_w_area(rmi_b)
+
+            # boiler to loop dict
+            loop_attach_boiler_dict = {}
+            for boiler in find_all("$.boilers[*]", rmi_b):
+                loop_id = getattr_(boiler, "boiler", "loop")
+                if loop_id not in loop_attach_boiler_dict.keys():
+                    loop_attach_boiler_dict[loop_id] = list()
+                loop_attach_boiler_dict[loop_id].append(boiler["id"])
+
+            # Initialize the variables
+            # The heating_loop_conditioned_zone_area will include sum of all connected zones and indirectly zone areas.
+            heating_loop_conditioned_zone_area = ZERO.AREA
+            # The connected zones list, zones in this list can be residential, nonresidential, mixed or semi-heated
+            loop_zone_list = []
+
+            for fluid_loop in find_all("$.fluid_loops[*]", rmi_b):
+                # Make sure heating loop, and its heating is supplied by a boiler(s)
+                if (
+                    getattr_(fluid_loop, "fluid_loops", "type") == FLUID_LOOP.HEATING
+                    and fluid_loop["id"] in loop_attach_boiler_dict.keys()
```

Comment:
Should the RDS be updated to include the logic of ensuring that the loop is attached to a boiler?